src/core/server/http/logging/get_payload_size.ts

```diff
@@ -17,8 +18,15 @@ const isBuffer = (src: unknown, res: Response): src is Buffer => {
   return !isBoom(res) && res.variety === 'buffer' && res.source === src;
 };
 const isFsReadStream = (src: unknown, res: Response): src is ReadStream => {
-  return !isBoom(res) && res.variety === 'stream' && res.source === src;
+  return (
+    !isBoom(res) &&
+    res.variety === 'stream' &&
+    res.source === src &&
+    res.source instanceof ReadStream
```

Comment:
suggestion but not a blocker: add an inline comment as to why we need to explicitly check the instance type:
```suggestion
    res.source instanceof ReadStream // prevent zlib's Gunzip streams through
```